pyomo/contrib/pynumero/interfaces/cyipopt_interface.py

```diff
@@ -328,24 +346,56 @@ def scaling_factors(self):
         return obj_scaling, x_scaling, g_scaling
 
     def objective(self, x):
-        self._set_primals_if_necessary(x)
-        return self._nlp.evaluate_objective()
+        try:
+            self._set_primals_if_necessary(x)
+            return self._nlp.evaluate_objective()
+        except PyNumeroEvaluationError:
+            # TODO: halt_on_evaluation_error option. If set, we re-raise the
+            # original exception.
```

Comment:
Can this TODO be removed? Doesn't this PR add the `halt_on_evaluation_error` option?

Reply:
Yes, that is out of date. Thanks for catching